pkg/netpol/eval/resources.go

```diff
@@ -531,8 +583,13 @@ func (pe *PolicyEngine) ConvertPeerNamedPort(namedPort string, peer Peer) (int32
 
 // AddPodByNameAndNamespace adds a new fake pod to:
 // the pe.podsMap in case of fake ingress-controller pods
-// or the pe.representativePeersList in case of exposure-analysis peers
-func (pe *PolicyEngine) AddPodByNameAndNamespace(name, ns string, nsLabels map[string]string) (Peer, error) {
+// or the pe.representativePeersMap in case of exposure-analysis peers
+func (pe *PolicyEngine) AddPodByNameAndNamespace(name, ns string, objLabels *k8s.SingleRuleLabels) (Peer, error) {
+	var nsLabels map[string]string
+	if objLabels != nil {
+		nsLabels = objLabels.NsLabels
+	}
```

Comment:
The map can be different from `nil` but with `len` that equals 0.. thus better check comparison of len to 0 instead of comparison to `nil` (if it's nil then also the len is 0).

Should not continue with this function if the len of input labels is 0?

Reply:
`objLabels` is not a map; it is a pointer to struct which eventually will include t2o maps (one for namespace labels an the other for pod labels) ; currently it may include only namespace labels 

so I check if this pointer is not nil; then assign `nsLabels` which is declared in line 588 (one line before the if statement) with the struct's `NsLabels`; otherwise we are creating a new pod without any namespace labels (like ingress-controller)

Reply:
ok